torch/csrc/autograd/custom_function.cpp

```diff
@@ -337,14 +338,13 @@ static optional_variable_list _process_backward_mode_ad(
         auto& grad_acc = dynamic_cast<AccumulateGrad&>(*grad_acc_fn);
         grad_acc.variable.reset();
       }
-      if (cdata) {
-        impl::rebase_history(var, {cdata, output_nr});
-      }
+      check_inplace(var, true);
+      impl::rebase_history(var, {cdata, output_nr});
     } else if (is_input) {
       TORCH_CHECK(!is_saved_and_setup_context, error_msg_input_returned_as_is)
       var = _view_as_self_with_no_grad(var, view_as_self_fn);
       impl::set_gradient_edge(var, {cdata, output_nr});
-    } else if (cdata) {
+    } else {
```

Comment:
the "if cdata" can be removed because is_differentiable=true guarantees non-null cdata

Reply:
Given the condition below:
    bool is_differentiable = cdata &&
        non_differentiable.count(out_tensor_impl) == 0 &&
        isDifferentiableType(var.scalar_type());

It's not true? :p